File: tools/calibrate/testdata/corpus/java/EventRouter.java
package example.net;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
import java.util.function.Consumer;

public final class EventRouter {
    private final Map<Class<?>, Consumer<Object>> handlers = new HashMap<>();

    public <T> void subscribe(Class<T> type, Consumer<? super T> handler) {
        Objects.requireNonNull(type, "type");
        Objects.requireNonNull(handler, "handler");
        handlers.put(type, event -> handler.accept(type.cast(event)));
    }

    public boolean publish(Object event) {
        Objects.requireNonNull(event, "event");
        Consumer<Object> handler = handlers.get(event.getClass());
        if (handler == null) {
            return false;
        }
        handler.accept(event);
        return true;
    }

    public int subscriptionCount() {
        return handlers.size();
    }
}

File: tools/calibrate/testdata/holdout/kotlin/EventRouter.kt
package holdout.events

import java.util.concurrent.ConcurrentHashMap

data class Event(val topic: String, val attributes: Map<String, String>, val body: String)
typealias EventHandler = (Event) -> Unit

class EventRouter {
    private val handlers = ConcurrentHashMap<String, MutableList<EventHandler>>()

    fun subscribe(topic: String, handler: EventHandler): AutoCloseable {
        require(topic.isNotBlank())
        val topicHandlers = handlers.computeIfAbsent(topic) { mutableListOf() }
        synchronized(topicHandlers) { topicHandlers += handler }
        return AutoCloseable {
            synchronized(topicHandlers) {
                topicHandlers.remove(handler)
                if (topicHandlers.isEmpty()) handlers.remove(topic, topicHandlers)
            }
        }
    }

    fun publish(event: Event): Int {
        val selected = buildList {
            handlers[event.topic]?.let { synchronized(it) { addAll(it) } }
            handlers["*"]?.let { synchronized(it) { addAll(it) } }
        }
        selected.forEach { handler -> handler(event) }
        return selected.size
    }
}
